### PR TITLE
Explore by folder in Codemod-first breakdown

### PR DESCRIPTION
**Pull request by hbjORbj**
### Linear: https://linear.app/intuita/issue/INT-843/explore-by-folder-in-codemod-first-breakdown

> - Allow user to explore by folder in Codemod-first breakdown
> - Consequently, we no longer display the list of individual jobs under each case

> New design is:
> - For each case, there is a root folder with subfolders. 
> - Each folder has "Commit" / "Discard" button.

### Result
https://user-images.githubusercontent.com/32841130/233623062-7e7bb585-e5ce-4a45-b050-d02b6b9b8d96.mov

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -405,12 +405,12 @@
 			"view/title": [
 				{
 					"command": "intuita.caseBreakdown",
-					"when": "view == intuitaMainWebviewExplorer || view == intuitaMainWebview",
+					"when": "false",
```
Comment (grzpab) [inline package.json:408]: hang on, then this command will be invisible right?

File: src/components/webview/MainWebviewProvider.ts [hunk 2]
```diff
@@ -118,42 +120,122 @@ export class IntuitaProvider implements WebviewViewProvider {
...
+				children,
 			};
 		}
 
 		if (element.kind === ElementKind.CASE) {
-			mappedNode = {
-				...mappedNode,
-				...this.__buildCaseTree(element),
-			};
+			if (this.__codemodMap.has(element.codemodName)) {
```
Comment (grzpab) [inline src/components/webview/MainWebviewProvider.ts:149]: this means the elements are up to date?

File: src/extension.ts [hunk 3]
```diff
@@ -1309,20 +1309,133 @@ export async function activate(context: vscode.ExtensionContext) {
...
 			'intuita.acceptFolder',
-			async (arg0, ...otherArgs) => {
-				const firstJobHash: string | null =
-					typeof arg0 === 'string' ? arg0 : null;
-				if (firstJobHash === null) {
-					throw new Error(
-						'Did not pass the jobHashes into the command.',
+			async (_arg0) => {
+				try {
+					const arg0: {
```
Comment (grzpab) [inline src/extension.ts:1314]: you can actually use a `io-ts` codec for that one

File: src/extension.ts [hunk 4]
```diff
@@ -1309,20 +1309,133 @@ export async function activate(context: vscode.ExtensionContext) {
...
+
+						if (job === null) {
+							continue;
+						}
+
+						stagedJobs.push({
+							hash: job.hash.toString(),
+							label: buildJobElementLabel(
+								job,
+								vscode.workspace.workspaceFolders?.[0]?.uri
```
Comment (grzpab) [inline src/extension.ts:1371]: this should be calculate before the for-each loop

File: src/extension.ts [hunk 5]
```diff
@@ -1309,20 +1309,133 @@ export async function activate(context: vscode.ExtensionContext) {
...
+							initialData,
+							viewColumn: vscode.ViewColumn.One,
+							webviewName: 'sourceControl',
+						},
+						messageBus,
+					);
+
+					await panelInstance.render();
+
+					const remotes = repositoryService.getRemotes();
```
Comment (grzpab) [inline src/extension.ts:1404]: `const remotes = repositoryService.getRemotes() ?? [];`
Though `getRemotes` should return an empty array by default.

File: package.json (hunk 1 above)
Comment (hbjORbj) [reply package.json:408]: yes, to my understanding, we don't need this command unless we provide options between caseBreakdown vs folderBreakdown.

File: src/components/webview/MainWebviewProvider.ts (hunk 2 above)
Comment (hbjORbj) [reply src/components/webview/MainWebviewProvider.ts:149]: This conditional never should be true. If it is true, that means we have `CASE` elements that have the same codemod name. Will leave this as a comment.

File: src/extension.ts (hunk 5 above)
Comment (DmytroHryshyn) [reply src/extension.ts:1404]: getRemotes already returns an empty array @grzpab

File: src/extension.ts (hunk 4 above)
Comment (DmytroHryshyn) [reply src/extension.ts:1371]: Can you explain? Here we just iterate jobs and build labels for each job

File: src/extension.ts (hunk 3 above)
Comment (hbjORbj) [reply src/extension.ts:1314]: addressed

**Review by DmytroHryshyn — APPROVED** [review]

**Review by grzpab — APPROVED** [review]

**Pull request merged**